controllers/volumehandler.go

```diff
@@ -368,6 +370,49 @@ func (h *sourceVolumeHandler) pvcFromSnap(l logr.Logger) (bool, error) {
 	return true, nil
 }
 
+func (h *sourceVolumeHandler) pvcForCache(l logr.Logger) (bool, error) {
+	h.resticCache = &v1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "restic-cache",
+			Namespace: h.Instance.Namespace,
+		},
+	}
+	logger := l.WithValues("pvc", nameFor(h.resticCache))
+
+	if h.Instance.Spec.Restic.LastPruned == nil {
```

Comment:
I don't understand why this check is here

Reply:
This is to check to avoid creating  `restic-cache pvc` on every reconciliation. 
On retrospecting, it should be `h.Instance.Status.LastSyncTime` instead of `h.Instance.Spec.Restic.LastPruned`
When the CR is applied for the first time, LastSyncTime will be null, so it is the first sync hence create restic cache. For all the future syncs LastSyncTime will have a value so it will not create `restic-cache pvc`

Reply:
Since you're using CreateOrUpdate, it's not going to delete/recreate. In fact, all the stuff inside the `h.resticCache.CreationTimestamp.IsZero()` test will only run if the PVC is actually being created. The only change that will be made on each reconcile is to ensure the controller reference is correct.